kserve-module/tests/e2e/conftest.py

```
@@ -49,6 +50,53 @@ class ClusterInfo:
+def _detect_openshift():
+    """Detect cluster type once; cached for the process lifetime."""
+    global _cluster_detection_cache
+    if _cluster_detection_cache is not None:
+        return _cluster_detection_cache
+
+    cli = "oc" if shutil.which("oc") else "kubectl"
+    if not shutil.which(cli):
+        _cluster_detection_cache = (False, cli)
+        return _cluster_detection_cache
+
+    result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)